runtimes/core/src/api/pvalue.rs

```diff
@@ -50,6 +50,18 @@ impl PValue {
             _ => None,
         }
     }
+
+    pub fn type_name(&self) -> &'static str {
+        match self {
```

Comment:
I would have expected we would introduce a new `PValue::Cookie` enum variant for cookies, so we don't need to treat it as an `Object` with string key lookups like `"maxAge"`

Reply:
I've updated to add a cookie variant!